sync/src/orphan_block_pool.rs

```diff
@@ -88,6 +90,31 @@ impl InnerPool {
                 .and_then(|blocks| blocks.get(hash).cloned())
         })
     }
+
+    /// cleanup expired blocks(epoch + EXPIRED_EPOCH < tip_epoch)
+    pub fn clean_expired_blocks(&mut self, tip_epoch: EpochNumber) -> Vec<packed::Byte32> {
+        let mut result = vec![];
+        let leaders: Vec<ParentHash> = self.leaders.iter().cloned().collect();
+        for hash in leaders.iter() {
+            if let Some(block) = self.get_block_by_parent(hash) {
+                if block.header().epoch().number() + EXPIRED_EPOCH < tip_epoch {
+                    // remove items in orphan pool and header map
+                    let descendants = self.remove_blocks_by_parent(&hash);
+                    for block in descendants {
+                        result.push(block.hash());
```

Comment:
use `expend` instead

Reply:
i just want the block hash, not entire block. (sounds too heavy)
then return the hash collection to callee function, it can free items in header.

result.extend( descendants.iter().hash()  )  <--for illustration, but it doesn't work.

Reply:
result.extend(descendants.iter().map(|a| a.hash()))

Reply:
got it! thanks....